WalletWasabi.Fluent/ViewModels/Wallets/Receive/ReceiveAddressViewModel.cs

```diff
@@ -1,106 +1,91 @@
-using System.Linq;
+using System.Collections.Generic;
 using System.Reactive;
-using System.Reactive.Disposables;
 using System.Reactive.Linq;
-using System.Threading;
 using System.Threading.Tasks;
 using System.Windows.Input;
-using Avalonia;
-using Gma.QrCodeNet.Encoding;
-using NBitcoin;
+using DynamicData;
 using ReactiveUI;
-using WalletWasabi.Blockchain.Analysis.Clustering;
-using WalletWasabi.Blockchain.Keys;
 using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Models.UI;
+using WalletWasabi.Fluent.Models.Wallets;
 using WalletWasabi.Fluent.ViewModels.Navigation;
-using WalletWasabi.Hwi;
 using WalletWasabi.Logging;
-using WalletWasabi.Wallets;
 
 namespace WalletWasabi.Fluent.ViewModels.Wallets.Receive;
 
 [NavigationMetaData(Title = "Receive Address")]
 public partial class ReceiveAddressViewModel : RoutableViewModel
 {
-	private readonly Wallet _wallet;
-	private readonly HdPubKey _model;
-
-	public ReceiveAddressViewModel(Wallet wallet, HdPubKey model)
+	public ReceiveAddressViewModel(UiContext uiContext, IWalletModel wallet, IAddress model, bool isAutoCopyEnabled)
 	{
-		_wallet = wallet;
-		_model = model;
-		Address = _model.GetP2wpkhAddress(_wallet.Network).ToString();
-		Labels = _model.Label;
-		IsHardwareWallet = _wallet.KeyManager.IsHardwareWallet;
-		IsAutoCopyEnabled = Services.UiConfig.Autocopy;
-
-		GenerateQrCode();
+		UiContext = uiContext;
+		Model = model;
+		Address = model.Text;
+		Labels = model.Labels;
+		IsHardwareWallet = wallet.IsHardwareWallet();
+		IsAutoCopyEnabled = isAutoCopyEnabled;
 
 		SetupCancel(enableCancel: false, enableCancelOnEscape: true, enableCancelOnPressed: true);
 
 		EnableBack = true;
 
-		CopyAddressCommand = ReactiveCommand.CreateFromTask(async () =>
-		{
-			if (Application.Current is { Clipboard: { } clipboard })
-			{
-				await clipboard.SetTextAsync(Address);
-			}
-		});
-
-		ShowOnHwWalletCommand = ReactiveCommand.CreateFromTask(async () => await OnShowOnHwWalletAsync(_model, _wallet.Network, _wallet.KeyManager.MasterFingerprint));
+		CopyAddressCommand = ReactiveCommand.CreateFromTask(() => UiContext.Clipboard.SetTextAsync(Address));
 
-		SaveQrCodeCommand = ReactiveCommand.CreateFromTask(async () => await OnSaveQrCodeAsync());
+		ShowOnHwWalletCommand = ReactiveCommand.CreateFromTask(ShowOnHwWalletAsync);
 
-		SaveQrCodeCommand.ThrownExceptions
+		var saveQrCodeCommand = ReactiveCommand.CreateFromTask(OnSaveQrCodeAsync);
+		saveQrCodeCommand.ThrownExceptions
 			.ObserveOn(RxApp.TaskpoolScheduler)
 			.Subscribe(ex => Logger.LogError(ex));
 
+		SaveQrCodeCommand = saveQrCodeCommand;
+		
 		NextCommand = CancelCommand;
+
+		QrCode = UiContext.QrCodeGenerator.Generate(model.Text);
+
+		if (IsAutoCopyEnabled)
+		{
+			CopyAddressCommand.Execute(null);
```

Comment:
I can see that this is here to be able to test the function. But now in the view, the `ExecuteOnOpen` is redundant as that will execute the command too.

What do you think about removing completely the `ExecuteOnOpen` property and logic from `AnimatedButton` and on the view just do `AnimateIcon="{Binding IsAutoCopyEnabled}" in the axaml`?
So the ViewModel will be responsible for copying but also the Control will give feedback about the copy.
It would be better if the feedback trigger and the actual copy execution would be in the same place, but because of the testability, this is good for now.

Reply:
It works and it's way nicer now. Thanks!